pkg/slacknotifier/slack_oauth.go

```diff
@@ -0,0 +1,52 @@
+package slacknotifier
+
+import (
+	"net/http"
+
+	"github.com/gofiber/fiber/v2"
+	"golang.org/x/oauth2"
+)
+
+var (
+	SlackBotClientID     = ""
+	SlackBotClientSecret = ""
+	oauthConfig          = &oauth2.Config{
+		ClientID:     SlackBotClientID,
+		ClientSecret: SlackBotClientSecret,
+		Endpoint: oauth2.Endpoint{
+			AuthURL:  "https://slack.com/oauth/v2/authorize",
+			TokenURL: "https://slack.com/api/oauth.v2.access",
+		},
+		Scopes: []string{
+			"chat:write",
+			"chat:write.public"},
+	}
+)
+
+// OauthHandler creates a handler for slack authentication
+func OauthHandler() fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		ctx := c.Context()
+
+		errStr := c.Params("error")
+		if errStr != "" {
+			c.Status(http.StatusUnauthorized)
+			_, err := c.WriteString(errStr)
```

Comment:
Should we return original errStr in such case or writing error?

Reply:
yes, if something happens from the slack side we should inform user.

Reply:
I meant, this method will return nil usually in case of non empty errStr, because c.WriteString will be succesful

Reply:
yes, you are right, good catch 👍

Reply:
actually it is the error of the handler func, we care about the request status here, so it should be fine